### ISSUE TITLE
Monitoring data import not working over SSL

### ISSUE THREAD

**Issue by inqueue** [elastic/support-diagnostics#350]
Attempting to push monitoring data to an ESS cluster with monitoring enabled.

```
./import-monitoring.sh --host $ESSENDPOINT --port 9243 --input monitoring-export-20191210-223252.tar.gz -u $EUSER -p $EPASS --ssl --clusterName testmon --indexName testmon
Using /usr/bin/java as Java Runtime
Using -Xms256m -Xmx2000m  for options.
Processing diagnosticInputs...
Deleted directory: /Users/<user>/bin/support-diagnostics-7.1.5/monitoring-export.
Creating temp directory: /Users/<user>/bin/support-diagnostics-7.1.5/monitoring-export
Configuring log file.
Extracting archive...
Processing: cluster_stats.json
Indexing document batch 0 to 250
Unexpected Execution Error
javax.net.ssl.SSLProtocolException: Broken pipe (Write failed)
	at sun.security.ssl.Alert.createSSLException(Alert.java:126) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:321) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:264) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:259) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:988) ~[?:?]
	at org.apache.http.impl.io.SessionOutputBufferImpl.streamWrite(SessionOutputBufferImpl.java:124) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.SessionOutputBufferImpl.write(SessionOutputBufferImpl.java:160) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:113) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:120) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.entity.StringEntity.writeTo(StringEntity.java:167) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:156) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:152) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:238) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:123) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72) ~[httpclient-4.5.10.jar:4.5.10]
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:57) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:79) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:68) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportProcessor.writeBatch(MonitoringImportProcessor.java:124) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportProcessor.process(MonitoringImportProcessor.java:93) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.util.ArchiveUtils.extractDiagnosticArchive(ArchiveUtils.java:100) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:46) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:21) [support-diagnostics-7.1.5.jar:7.1.5]
Caused by: java.net.SocketException: Broken pipe (Write failed)
	at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:?]
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:150) ~[?:?]
	at sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:320) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:983) ~[?:?]
	... 23 more
Error processing JSON event for {}.
java.lang.RuntimeException: Unexpected exception
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:63) ~[support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:79) ~[support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:68) ~[support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportProcessor.writeBatch(MonitoringImportProcessor.java:124) ~[support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportProcessor.process(MonitoringImportProcessor.java:93) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.util.ArchiveUtils.extractDiagnosticArchive(ArchiveUtils.java:100) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:46) [support-diagnostics-7.1.5.jar:7.1.5]
	at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:21) [support-diagnostics-7.1.5.jar:7.1.5]
0 events written from cluster_stats.json
```

Looks like it is going to work in the clear, though I received an I/O exception on my first try:

```
./import-monitoring.sh --host $ESENDPOINT --port 9200 --input monitoring-export-20191210-223252.tar.gz -u $EUSER -p $EPASS --clusterName testmon --indexName testmon
Using /usr/bin/java as Java Runtime
Using -Xms256m -Xmx2000m  for options.
Processing diagnosticInputs...
Deleted directory: /Users/<user>/bin/support-diagnostics-7.1.5/monitoring-export.
Creating temp directory: /Users/<user>/bin/support-diagnostics-7.1.5/monitoring-export
Configuring log file.
Extracting archive...
Processing: cluster_stats.json
Indexing document batch 0 to 250
I/O exception (java.net.SocketException) caught when processing request to {}->http://d5d6aed...1fc.us-east-1.aws.found.io:9200: Broken pipe (Write failed)
Retrying request to {}->http://d5d6aed...1fc.us-east-1.aws.found.io:9200
```

The cluster was configured at 1g. Trying the slider at 4g.

**Comment by nemonster** [comment]
@inqueue What version is the cluster you're trying to index to?

**Comment by inqueue** [comment]
> @inqueue What version is the cluster you're trying to index to?

I am trying to send it to a 7.5.0 cluster.

**Comment by nemonster** [comment]
Is it possible for me to give this a try myself and run it in debug mode?

**Comment by nemonster** [comment]
Appears to be resolved so closing for now.

**Comment by PhaedrusTheGreek** [comment]
I'm having the same issue with Cloud/SSL.    @inqueue what did you do to get past this?  The same error is happening with or without the `--ssl` flag, so I wonder if it's not being picked up.

**Comment by inqueue** [comment]
@PhaedrusTheGreek As far as I know import to Cloud ESS is actually not yet supported so I avoided ESS  entirely.

**Comment by perrinal** [comment]
II tried today and got the same error it would be nice to have this working as monitoring uploaded in ESS is easier to share.

**Comment by bczifra** [comment]
@nemonster this is not resolved. the three folks above who commented ran into this issue, as did I. Can this issue please be reopened?

**Comment by nemonster** [comment]
@bczifra This doesn't have anything to do with SSL. It has to do with ESS. I'm looking into it at the present. For now, if you want to use the import, do it on a local instance. It will work with or without SSL.

**Comment by inqueue** [comment]
The error presented itself as an SSL issue and I assumed import would "just work" to ESS. How about we rephrase the title to `Monitoring data import not working to ESS` and reopen?

**Comment by eedugon** [comment]
This was a real issue (and easy to reproduce), but we have closed it (and similar reports) and fixed in 9.0.0 without details...

Anyway this looks fixed in 9.0.0 pre-release, check this nice comment for more info and details on how to execute the import:
https://github.com/elastic/support-diagnostics/issues/457#issuecomment-780844011
